src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -0,0 +1,249 @@
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.InternalBibtexFields;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.fxmisc.easybind.EasyBind;
+import org.fxmisc.flowless.VirtualizedScrollPane;
+import org.fxmisc.richtext.CodeArea;
+
+public class SourceTab extends EntryEditorTab {
+
+    private static final Log LOGGER = LogFactory.getLog(SourceTab.class);
+    private final BibDatabaseMode mode;
+    private final BibEntry entry;
+    private CodeArea codeArea;
+
+    public SourceTab(BibDatabaseMode mode, BibEntry entry) {
+        this.mode = mode;
+        this.entry = entry;
+        this.setText(Localization.lang("%0 source", mode.getFormattedName()));
+        this.setTooltip(new Tooltip(Localization.lang("Show/edit %0 source", mode.getFormattedName())));
+        this.setGraphic(IconTheme.JabRefIcon.SOURCE.getGraphicNode());
+    }
+
+    private static String getSourceString(BibEntry entry, BibDatabaseMode type) throws IOException {
+        StringWriter stringWriter = new StringWriter(200);
+        LatexFieldFormatter formatter = LatexFieldFormatter
+                .buildIgnoreHashes(Globals.prefs.getLatexFieldFormatterPreferences());
+        new BibEntryWriter(formatter, false).writeWithoutPrependedNewlines(entry, stringWriter, type);
+
+        return stringWriter.getBuffer().toString();
+    }
+
+    private Node createSourceEditor(BibEntry entry, BibDatabaseMode mode) {
+        codeArea = new CodeArea();
+        codeArea.setWrapText(true);
+        //codeArea.(Font.font("Monospaced", Globals.prefs.getInt(JabRefPreferences.FONT_SIZE)));
+        EasyBind.subscribe(codeArea.focusedProperty(), focused -> {
+            if (!focused) {
+                storeSource();
+            }
+        });
+
+        try {
+            String srcString = getSourceString(entry, mode);
+            codeArea.appendText(srcString);
+        } catch (IOException ex) {
+            codeArea.appendText(ex.getMessage() + "\n\n" +
+                    Localization.lang("Correct the entry, and reopen editor to display/edit source."));
+            codeArea.setEditable(false);
+            LOGGER.debug("Incorrect entry", ex);
+        }
+
+        return new VirtualizedScrollPane<>(codeArea);
+    }
+
+    @Override
+    public boolean shouldShow() {
+        return true;
+    }
+
+    @Override
+    protected void initialize() {
+        this.setContent(createSourceEditor(entry, mode));
+    }
+
+    /**
+     * NOTE: This method is only used for the source panel, not for the
+     * other tabs. Look at EntryEditorTab for the setup of text components
+     * in the other tabs.
+     */
+    private void setupJTextComponent(JTextComponent textComponent) {
+        /*
+        // TODO: Set up key bindings and focus listener for the code editor.
```

Comment:
Agreed. The entry editor specific key bindings are now activated again and they work as expected. There are still issues with the key bindings, especially when it comes to more global ones - but this is already tracked as https://github.com/JabRef/jabref/issues/2855.